spec.html

```diff
@@ -5534,6 +5534,16 @@ <h1>IsStringPrefix ( _p_, _q_ )</h1>
       </emu-note>
     </emu-clause>
 
+    <emu-clause id="sec-hasidentity" aoid="HasIdentity">
+      <h1>HasIdentity ( _argument_ )</h1>
```

Comment:
```suggestion
      <h1>HasUniqueIdentity ( _argument_ )</h1>
```
🚲 🏠 ?

Reply:
IsUnforgeable?

Reply:
"unforgeable" means something very different to me, not dealing with identity but with observable characteristics of a value.

Reply:
I liked @bakkot's suggestion, so I have adopted it here. Do you have a particular concern with it?

Reply:
I view all primitives as having identity, and the “unique identity” phrasing you used in the prose seemed more precise to me.

Reply:
I view all primitives as having identity, and the “unique identity” phrasing you used in the prose seemed more precise to me.

Reply:
I am confused by @ljharb's use of the word "identity", which is quite opposite of my understanding of how it's used usually.

"Identity" is the thing that distinguishes encapsulated things as different from one another. JS encapsulates objects and thus objects have identity: `({a:42}) !== ({a:42})`. The two objects there have the same structure, but different identities. JS does not encapsulate numbers and strings: `"foo" === "foo"`, and `3 === 3`.

It is wrong to teach "identity" as the reason that `3 === 3`. `3 === 3` precisely because we don't encapsulate numbers. Python is the most infamous here, I think, in that it [encapsulates number up to 256](https://stackoverflow.com/questions/306313/is-operator-behaves-unexpectedly-with-integers) due to implementation reasons.

Reply:
I would be happy with @claudepache's suggestions of "collectable" or "unforgeable" to describe predicate, as well as "has identity". I think we can defer settling on a conclusion here until we're gettijg ready to merge this PR.